docs/source/miscellaneous/doc/adr/0014-cassandra-targeted-sweep.md

```diff
@@ -0,0 +1,321 @@
+and start row.
+
+Note that there are corner cases where the sweep timestamp may be mistaken: a read-only transaction that has been running
+for longer than one hour (which is also subject to clock drift), or when a write transaction loses its lock while committing. In
+tables that allow read-only transactions we must therefore defensively write a **garbage deletion sentinel**, which is an empty
+value at a timestamp of -1, i.e., before any valid start timestamp. If a read-only transaction encounters a sentinel, that
+signalizes there could have been a historic version it should be able to read that may have been deleted, and the transaction must
+therefore abort. If a table allows read-only transactions, and therefore requires sentinels, is defined by the table's
+**sweep strategy**: `CONSERVATIVE`, which allows read-only transactions and requires sentinels, and `THOROUGH`, which does not.
+
+Over time we have identified a number of issues with the architecture and implementation of legacy sweep as outlined below.
+
+#### Legacy Sweep is Slow
+
+Even if a table has no historic versions of cells that can be swept, we still have to scan through all historic versions of
+all cells in the table, which can take weeks in extreme cases. The performance only gets worse as more data is written to the
+KVS, increasing the number of entries legacy sweep must iterate through. This is a particularly large problem for users with tables
+whose access patterns mandate that they must be swept regularly for performance or stability reasons. If background sweep is busy
+for weeks sweeping some other large table, we must resort to manual sweeps to avoid performance and stability degradation. This is
+obviously error-prone and subject to random failures, since if the manual sweep is interrupted for any reason, it will not be
+automatically retried.
+
+Moreover, legacy sweep depends on complicated heuristics to decide which table to sweep next. Given that some tables must be swept
+frequently, and the slowness of legacy sweep, significant developer time must be spent tweaking the heuristic to produce the desired
+effect.
+
+#### Legacy Sweep can Exert Significant Pressure on the Underlying KVS
+
+Scanning through a table to find historic versions of cells can in some cases cause significant pressure on the underlying KVS,
+in particular for Cassandra. Even though we are only interested in the start and commit timestamp of the transaction in which the
+writes were performed, it is our understanding that Cassandra internally still loads the contents of the cell in memory regardless.
+
+#### Legacy Sweep can Get Stuck
+
+For tables that regularly have new rows added in increasing lexicographical order (for example, tables keyed on `FIXED` or
+`VAR_LONG`), legacy sweep can end up sweeping the table indefinitely as each iteration will discover a new row, therefore never
+declaring the table as fully swept.
```

Comment:
Maybe obvious, but we should state the consequence of this - no other tables are swept at all (prior to multi-sweep), and with multi-sweep, one of the sweep threads spins forever on this table.